src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -714,75 +614,91 @@ protected override void EndProcessing()
                     new ErrorRecord(
                         new PSNotSupportedException(AddTypeStrings.CannotDefineNewType), "CannotDefineNewType", ErrorCategory.PermissionDenied, null));
             }
+        }
 
-            // Generate an error if they've specified an output
-            // assembly type without an output assembly
-            if (String.IsNullOrEmpty(outputAssembly) && outputTypeSpecified)
```

Comment:
`OutputType` is translated in CompilationOptions. Since we have ExtendedOptions parameter we can catch tons conflicts - we'll get runtime errors. I guess we won't be able to identify all conflicts- it is better to delegate this to Roslyn.

Reply:
Then why do you check this case: `Add-Type $code -CompileOnly` without `-OutputAssemby`?
They look to me are the same situation.

Reply:
Fixed.